lib/saluki-components/src/sources/heartbeat/mod.rs

```diff
@@ -0,0 +1,105 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_context::Context;
+use saluki_core::components::{sources::*, ComponentContext};
+use saluki_core::topology::OutputDefinition;
+use saluki_error::GenericError;
+use saluki_event::{metric::Metric, DataType, Event};
+use tokio::{select, time::interval};
+use tracing::{debug, error};
+
+/// Configuration for a source that doesn't produce any events or optionally emits heartbeat metrics.
```

Comment:
```suggestion
/// Heartbeat source.
///
/// Emits a "heartbeat" metric on a configurable interval.
```

Since users generally only ever interact with the component configuration type (when building the topology), we strive to put the human-friendly name/description of a component in the doc comments of that configuration type itself.